corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -50,22 +50,29 @@ hqDefine("data_dictionary/js/data_dictionary", [
 
                     var propObj = propertyListItem(prop.name, prop.label, false, group.name, self.name, prop.data_type,
                         prop.description, prop.allowed_values, prop.fhir_resource_prop_path, prop.deprecated,
-                        prop.removeFHIRResourcePropertyPath, isGeoCaseProp, prop.is_safe_to_delete, prop.id);
-                    propObj.description.subscribe(changeSaveButton);
-                    propObj.label.subscribe(changeSaveButton);
-                    propObj.fhirResourcePropPath.subscribe(changeSaveButton);
-                    propObj.dataType.subscribe(changeSaveButton);
-                    propObj.deprecated.subscribe(changeSaveButton);
-                    propObj.deleted.subscribe(changeSaveButton);
-                    propObj.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
+                        prop.removeFHIRResourcePropertyPath, isGeoCaseProp, prop.is_safe_to_delete, prop.id, prop.index, groupObj.name());
+                    subscribePropObservable(propObj, propObj.description, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.label, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.fhirResourcePropPath, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.dataType, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.deprecated, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.deleted, changeSaveButton);
+                    subscribePropObservable(propObj, propObj.removeFHIRResourcePropertyPath, changeSaveButton);
                     propObj.allowedValues.on('change', changeSaveButton);
+                    propObj.allowedValues.on('change', propObj.allowedValuesChanged);
```

Comment:
Can you help understand how this is different from `valChanged` function?

Reply:
For `allowedValues` we use a `uiElementKeyValueList` component. When it fires the `change` event [here](https://github.com/dimagi/commcare-hq/blob/35379ef54421e9271ab6b0126b9067295b936b50/corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/bootstrap3/ui-element-key-val-list.js#L73) it doesn't pass any old/new values, so if we use `valChanged` this will always default to `false` in the conditional. So I just followed the same pattern as the old code where any fires to the `change` event will result in `hasChanges` being `true` for `allowedValues`.